app/controllers/dashboard-view-model.js

```diff
@@ -59,6 +62,52 @@ module.exports = class DashboardViewModel {
     ];
   }
 
+  dashboardTimeline(jobs) {
+    let currentStatus = 'unknown';
+
+    const priority = timelinePriorityRules.priority(jobs);
+    const text = timelineTextRules.text(jobs);
+
+    const statusList = Object.keys(priority);
+    const indexOfFirstDefaultPriority = statusList.map(
+      status => priority[status]).findIndex(p => p === 'default');
+
+    if (indexOfFirstDefaultPriority > 0) {
+      currentStatus = statusList[indexOfFirstDefaultPriority - 1];
+    }
+
+    return statusList.map((status, index) => (
+      {
+        // eslint-disable-next-line no-underscore-dangle
+        heading: i18n.__(`timeline.status.${status}`),
+        class: this.buildStatusClass(index, status, currentStatus, statusList, priority),
+        message: text[status],
+      }
+    ));
+  }
+
+  buildStatusClass(index, status, currentStatus, statusList, priority) {
+    const classArray = [];
+
+    if (priority[status] === singleStatusRules.PRIORITY.HIGH) {
+      classArray.push('timeline__item--highlight');
+    } else {
+      classArray.push('timeline__item--default');
+    }
+
+    if (index === 0) {
+      classArray.push('timeline__item--start');
+    } else if (index === statusList.length - 1) {
```

Comment:
`else if` or just `if`? can there be row that has both `timeline__item--start` and `timeline__item--finish`?